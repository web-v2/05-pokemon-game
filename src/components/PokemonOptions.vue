<template>
  <div class="options-container">
    <ul v-if="mostrar">
        <li v-for="pk in pokemons" :key="pk.id" 
        v-on:click="inactivarOpciones"         
        @click="$emit('selectionPokemon', pk.id)">{{ pk.name }}</li>
    </ul>
  </div>
</template>

<script>
export default {
    props: {
        pokemons: {
            type: Array,
            required: true
        }
    },
    data() {
        return {
            mostrar: true
        }
    },
    methods:{
         inactivarOpciones(){
            this.mostrar = false;
        }
    }
}
</script>

<style scoped>
/* Pokemon Options */
ul {
    list-style-type: none;
}
li {
    background-color: white;
    border-radius: 5px;
    border: 1px solid rgba(0, 0, 0, 0.2);
    cursor: pointer;
    margin-bottom: 12px;
    width: 190px;
}

li:hover {
    background-color: rgba(0, 0, 0, 0.05);
}

.options-container {
    display: flex;
    justify-content: center;
}
</style>