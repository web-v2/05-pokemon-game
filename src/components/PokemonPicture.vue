<template>
  <div class="pokemon-container">
    <img v-if="!showPokemon" 
         :src="imgSrc" 
         class="hidden-pokemon"
         alt="pokemon">
    <img v-else 
         :src="imgSrc" 
         class="fade-in"
         alt="pokemon">         
  </div>
</template>

<script>
export default {
    props: {        
        pokemonId: {
            type: Number,
            required: true
        },        
        showPokemon:{ 
            type: Boolean,
            required: true,
            default: false
        }
    },
    data() {
        return {
            Id: (this.pokemonId <= 0)?1:this.pokemonId
        }
    },
    computed:{        
        imgSrc(){    
            console.log(this.Id);        
            return `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${this.Id}.svg`
        }
    }
}
</script>

<style scoped>
/* Pokemon Picture */
.pokemon-container {
    height: 200px;
}
img {
  height: 200px;
  position: absolute;
  width: 100vw;
  margin-left: -50vw;
  user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-drag: none;
  -webkit-user-select: none;
}
.hidden-pokemon {
    filter: brightness(0);
}
</style>