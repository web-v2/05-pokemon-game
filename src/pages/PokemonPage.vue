<template>
  <div>
    <h1>¿Quién es este Pokemon?</h1>
    <PokemonPictureVue :pokemonId="250" :showPokemon="true"/>
    <PokemonOptionsVue :pokemons="pokemonArr"/>    
  </div>
</template>

<script>
import PokemonPictureVue from '@/components/PokemonPicture.vue'
import PokemonOptionsVue from '@/components/PokemonOptions.vue'

import getPokemonOptions from '@/helpers/getPokemonOptions'
//console.log(getPokemonOptions());

export default {
  components: { PokemonPictureVue, PokemonOptionsVue },
  data(){
    return{
        pokemonArr: []
    }
  },
  methods: { 
    async mixPokemonsArray(){
      this.pokemonArr = await getPokemonOptions();      
    }
  },
  mounted(){
    this.mixPokemonsArray()
    //console.log('Mounted');
  }
}
</script>