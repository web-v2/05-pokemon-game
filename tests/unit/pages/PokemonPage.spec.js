import {shallowMount, mount} from '@vue/test-utils'
import PokemonPage from '@/pages/PokemonPage'
import PokemonPicture from '@/components/PokemonPicture'
import PokemonOptions from '@/components/PokemonOptions'

import { pokemonsMock } from '../mocks/pokemons.mock'

describe('PokemonPage Component', () => {
    let wrapper
    beforeEach(() => {
        wrapper = shallowMount(PokemonPage)
    })

    test('Debe de hacer match con el snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })

    test('Debe de llamar el mixedPokemonsArr al montar ', () => {
        const mixPokemonsArraySpy = jest.spyOn(PokemonPage.methods, 'mixPokemonsArray')
        const wrapper = shallowMount(PokemonPage)
        expect(mixPokemonsArraySpy).toHaveBeenCalled()
    })

    test('Debe de hacer match con el snapshot cuando carguen los pakemons', () => {
        const wrapper = shallowMount(PokemonPage, {
            data(){
                return{
                    pokemonArr:   pokemonsMock,
                    pokemon:      pokemonsMock[0],
                    showPokemon:  false,
                    showAnswer:   false,
                    message:      '',
                    resultado:    '',
                    verify:       false,
                    ganados:      0,
                    perdidos:     0
                }
              }
        })
        expect(wrapper.html()).toMatchSnapshot()
    })

    test('Debe de mostrar los components PokemonPicture y PokemonOptions', () => {
        const wrapper = shallowMount(PokemonPage, {
            data(){
                return{
                    pokemonArr:   pokemonsMock,
                    pokemon:      pokemonsMock[0],
                    showPokemon:  false,
                    showAnswer:   false,
                    message:      '',
                    resultado:    '',
                    verify:       false,
                    ganados:      0,
                    perdidos:     0
                }
              }
        })
        const picture = wrapper.find('pokemon-picture-vue-stub')
        const options = wrapper.find('pokemon-options-vue-stub')

        expect(picture.exists()).toBeTruthy()
        expect(options.exists()).toBeTruthy()
        expect(picture.attributes('pokemonid')).toBe('1')
        expect(options.attributes('pokemons')).toBeTruthy()

    })

    test('Pruebas con checkAnswer', async() => {
        const wrapper = shallowMount(PokemonPage, {
            data(){
                return{
                    pokemonArr:   pokemonsMock,
                    pokemon:      pokemonsMock[0],
                    showPokemon:  false,
                    showAnswer:   false,
                    message:      '',
                    resultado:    '',
                    verify:       false,
                    ganados:      0,
                    perdidos:     0
                }
              }
        })
        
        await wrapper.vm.checkAnswer(1)
        expect(wrapper.find('h2').exists()).toBeTruthy()
        expect(wrapper.vm.showPokemon).toBe(true)
        expect(wrapper.find('h2').text()).toBe(`Correcto, es ${pokemonsMock[0].name}`)
        await wrapper.vm.checkAnswer(15)
        expect(wrapper.vm.message).toBe(` Oops, era ${pokemonsMock[0].name}`)
    })
})