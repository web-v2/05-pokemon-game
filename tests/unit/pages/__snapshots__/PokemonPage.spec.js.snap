// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPage Component Debe de hacer match con el snapshot 1`] = `
<div class="card">
  <div class="container">
    <h1>Espere por favor...</h1>
  </div>
</div>
`;

exports[`PokemonPage Component Debe de hacer match con el snapshot cuando carguen los pakemons 1`] = `
<div class="card">
  <div class="container">
    <div>
      <h1>¿Quién es este pokémon?</h1>
      <pokemon-picture-vue-stub pokemonid="1" showpokemon="false"></pokemon-picture-vue-stub>
      <pokemon-options-vue-stub pokemons="[object Object],[object Object],[object Object],[object Object]"></pokemon-options-vue-stub>
      <p>Acertados: <strong>0</strong> Errados: <strong>0</strong></p>
      <!--v-if-->
      <h3>Desarrollado por Samir V2 del Curso de Vue.js de <a href="https://fernando-herrera.com/#/" target="_blank">Fernando Herrera</a>.</h3>
    </div>
  </div>
</div>
`;
